features/filesystem/fat/ChaN/ff.h

```diff
@@ -93,9 +93,7 @@ typedef struct {
 	WORD	id;				/* Volume mount ID */
 	WORD	n_rootdir;		/* Number of root directory entries (FAT12/16) */
 	WORD	csize;			/* Cluster size [sectors] */
-#if FF_MAX_SS != FF_MIN_SS
 	WORD	ssize;			/* Sector size (512, 1024, 2048 or 4096) */
```

Comment:
Why is the conditional pre-processing removed here?

Reply:
This is in 694b9ad11c. The `FATFileSystem` code uses `ssize` from this struct directly to determine the sector size of the filesystem for calculations, [here](https://github.com/Eosis/mbed/blob/694b9ad11c5f9a85bdef6e2033d8abf5bc9213f7/features/filesystem/fat/FATFileSystem.cpp#L544-L546). If we are in the situation where `FF_MAX_SS == FF_MIN_SS`, then `FATFileSystem.cpp` will not compile because the variable is not in the struct.

This is dealt with in `ff.cpp` through the use of a macro, [here](https://github.com/Eosis/mbed/blob/694b9ad11c5f9a85bdef6e2033d8abf5bc9213f7/features/filesystem/fat/ChaN/ff.cpp#L228-L232).

I felt there were two options:
1) Move the macro from `ff.cpp` to `ff.h`, change the `FATFileSystem` code to use the new macro (it should be visible in `FATFileSystem.cpp` as it includes `ff.h` along the way).
2) Remove the check, having `ssize` defined in the struct for all sector sizes.

The latter seemed less intrusive (following the mbed contribution guidelines) so I went with this, accepting that there would be an increase in the memory footprint for the filehandle of `sizeof(WORD)`. What do you reckon?

Reply:
@Eosis - Thanks for detailed explanation, increase of `sizeof(WORD)` should not be big of problem.
But I believe option 1 is good approach, and you can add that fix to ChanFs repo as well.

@geky - What are your thoughts on this?